### PR TITLE
FIT23: Link users with weights

### PR DESCRIPTION
**Pull request by PiotrMichalak1**

### CODE REVIEW

File: database/changelog.yml [hunk 1]
```diff
@@ -3,3 +3,5 @@ databaseChangeLog:
       file: scripts/001-FIT-3-create_weights_table.sql
   - include:
       file: scripts/002-FIT-11-create_users_table.sql
+  - include:
+      file: scripts/003-FIT-23-link_users_with_weights.sql
```
[the remark this answers is not in the document]
Comment (PiotrMichalak1) [reply database/changelog.yml:7]: Good idea, but as we are altering weights table maybe we should put: `003-FIT-23-update_weights_table_with_user_id_foreign_key`

[the remark this answers is not in the document]
Comment (pd-dudkiewicz) [reply database/changelog.yml:7]: yes, that's what I meant actually :)

File: src/test/java/com/fittracker/fittracker/controller/AuthenticationControllerIntegrationTest.java [hunk 2]
```diff
@@ -44,8 +48,8 @@ protected String getEndpoint() {
 
     @BeforeEach
     void beforeEach() {
+        weightRepository.deleteAll();
```
[the remark this answers is not in the document]
Comment (PiotrMichalak1) [reply src/test/java/com/fittracker/fittracker/controller/AuthenticationControllerIntegrationTest.java:51]: https://dev.to/henrykeys/don-t-use-transactional-in-tests-40eb

Due to the source:

"Well, it turns out that @Transactional wraps the contained code within a transaction — which maintains the database session opened across all code contained in the transaction, which is needed for successful lazy loading of collections."

File: src/test/java/com/fittracker/fittracker/controller/AuthenticationControllerIntegrationTest.java [hunk 3]
```diff
@@ -64,7 +57,7 @@ void givenUsernameNorEmailExists_shouldReturnRegisterResponse() throws Exception
                     .andExpect(status().isCreated())
                     .andReturn().getResponse().getContentAsString();
 
-            var expected = new RegisterResponse(UUID.randomUUID(), "user", "user@example.com");
+            var expected = new RegisterResponse(UUID.fromString("948cc727-68e5-455c-ab6d-942e585bde0d"), "user", "user@example.com");
```
Comment (pd-dudkiewicz) [inline src/test/java/com/fittracker/fittracker/controller/AuthenticationControllerIntegrationTest.java:60]: let's just put the UUID object into private static final field

File: src/test/java/com/fittracker/fittracker/controller/AuthenticationControllerIntegrationTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (pd-dudkiewicz) [reply src/test/java/com/fittracker/fittracker/controller/AuthenticationControllerIntegrationTest.java]: it's not a problem now, I can see you've added it, so resolving the comment

File: src/test/java/com/fittracker/fittracker/request/WeightRequestTest.java [hunk 4]
```diff
@@ -1,32 +1,24 @@
...
         WeightRequest weightRequest = new WeightRequest(testDate, testValue);
-        var expectedResult = new Weight(testDate, testValue, UUID.randomUUID());
+        var expectedResult = new Weight(testDate, testValue);
 
-        try (MockedStatic<SecurityHelper> mockedStatic = mockStatic(SecurityHelper.class)) {
-            mockedStatic.when(SecurityHelper::getUserId).thenReturn(mockedUserId);
+        var result = weightRequest.toWeight();
 
-            var result = weightRequest.toWeight();
+        assertThat(result).usingRecursiveComparison().ignoringFields("userId").isEqualTo(expectedResult);
```
Comment (pd-dudkiewicz) [inline src/test/java/com/fittracker/fittracker/request/WeightRequestTest.java:21]: we don't need to ignore userId

File: src/test/java/com/fittracker/fittracker/controller/AuthenticationControllerIntegrationTest.java [hunk 5]
```diff
@@ -28,8 +28,13 @@ public class AuthenticationControllerIntegrationTest extends BaseIntegrationTest
     @Autowired
     private UserRepository userRepository;
 
+    @Autowired
+    private WeightRepository weightRepository;
```
Comment (pd-dudkiewicz) [inline src/test/java/com/fittracker/fittracker/controller/AuthenticationControllerIntegrationTest.java:32]: this is not needed

**Review by pd-dudkiewicz — APPROVED** [review]

**Pull request merged**